drivers/gles3/rasterizer_storage_gles3.cpp

```diff
@@ -3302,31 +3303,51 @@ void RasterizerStorageGLES3::mesh_add_surface(RID p_mesh, uint32_t p_format, VS:
 
 				if (p_format & VS::ARRAY_FLAG_USE_16_BIT_BONES) {
 					attribs[i].type = GL_UNSIGNED_SHORT;
-					stride += 8;
+					stride += p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS ? 16 : 8;
 				} else {
 					attribs[i].type = GL_UNSIGNED_BYTE;
-					stride += 4;
+					stride += p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS ? 8 : 4;
 				}
 
 				attribs[i].normalized = GL_FALSE;
 				attribs[i].integer = true;
 
+				if (p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS) {
+					attribs[VS::ARRAY_MAX+0] = attribs[i];
+					attribs[VS::ARRAY_MAX+0].index = 8;
+					attribs[VS::ARRAY_MAX+0].offset = attribs[i].offset + (p_format & VS::ARRAY_FLAG_USE_16_BIT_BONES ? 8 : 4);
+				} else {
+					attribs[VS::ARRAY_MAX+0].enabled = false;
+					attribs[VS::ARRAY_MAX+0].integer = false;
+				};
+
 			} break;
 			case VS::ARRAY_WEIGHTS: {
 
+				int count = p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS ? 8 : 4;
 				attribs[i].size = 4;
 
 				if (p_format & VS::ARRAY_COMPRESS_WEIGHTS) {
 
 					attribs[i].type = GL_UNSIGNED_SHORT;
-					stride += 8;
+					stride += p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS ? 16 : 8;
 					attribs[i].normalized = GL_TRUE;
 				} else {
 					attribs[i].type = GL_FLOAT;
-					stride += 16;
+					stride += p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS ? 32 : 16;
 					attribs[i].normalized = GL_FALSE;
 				}
 
+				if (p_format & VS::ARRAY_FLAG_USE_8_WEIGHTS) {
+					attribs[VS::ARRAY_MAX+1] = attribs[i];
+					attribs[VS::ARRAY_MAX+1].index = 9;
+					attribs[VS::ARRAY_MAX+1].offset = attribs[i].offset + (p_format & VS::ARRAY_COMPRESS_WEIGHTS ? 8 : 16);
+				} else {
+					attribs[VS::ARRAY_MAX+1].enabled = false;
+					attribs[VS::ARRAY_MAX+1].integer = false;
+				};
+
+
```

Comment:
The code bellow here (ARRAY_INDEX) really breaks your logic, because INDEX is MAX-1 and is executed after bone and weight, so you will need to do a more harcoded approach to setting those.